src/encoding-common.jl

```diff
@@ -120,43 +115,32 @@ end
 
 function encode(float::Union{Float64, Float32, Float16}, bytes::Array{UInt8, 1})
     if typeof(float) == Float64 && float == Float32(float)
-        const float_bytes = hex2bytes(num2hex(Float32(float)) )
+        x = Float32(float)
+        float_bytes = hex2bytes(string(reinterpret(Unsigned,x),base=16,pad=sizeof(x)*2) )
     else
-        const float_bytes = hex2bytes(num2hex(float))
+        float_bytes = hex2bytes(string(reinterpret(Unsigned,float),base=16,pad=sizeof(float)*2))
```

Comment:
If instead of using an array of bytes, you used an `IOBuffer`, you could just `write(buffer, hton(float))`.